runtime/monotouch-debug.m

```diff
@@ -1486,10 +1486,11 @@ int monotouch_debug_connect (NSMutableArray *ips, int debug_port, int output_por
 	unsigned char sockaddr[sizeof (struct sockaddr_in6)];
 	struct sockaddr_in6 *sin6 = (struct sockaddr_in6 *) sockaddr;
 	struct sockaddr_in *sin = (struct sockaddr_in *) sockaddr;
-	int family, waiting, len, rv, i;
-	int ip_count = [ips count];
+	int family, waiting, len, rv;
+	unsigned long i;
+	unsigned long ip_count = [ips count];
 	const char *family_str;
-	int connected = -1;
+	unsigned long connected = -1;
```

Comment:
I need to look at the logic of connected..

Reply:
Yes, as Whitney says, assigning a negative value to an unsigned variable is not ideal (it might work out just fine in the end, but at the very least it can be confusing for any future developers who have to debug this code).